Source/Scene/GlobeSurfaceShaderSet.js

```diff
@@ -260,8 +315,24 @@ define([
             }
         }
 
+        var pickShaderPrograms = this._pickShaderPrograms;
+        for (flags in pickShaderPrograms) {
+            if (pickShaderPrograms.hasOwnProperty(flags)) {
+                shader = pickShaderPrograms[flags];
+                shader.destroy();
+            }
+        }
+
+        var shadowCastPrograms = this._shadowCastPrograms;
+        for (flags in shadowCastPrograms) {
+            if (shadowCastPrograms.hasOwnProperty(flags)) {
```

Comment:
Also longer-term, the pick shader can likely be the same as the shadow cast shader since all it wants to do is render depth for opaque geometry.